### PR TITLE
enable async processing in pinot broker query api

### PR DESCRIPTION
**Pull request by jamesyfshao**
right now pinot broker query api issue a blocking call when it try to fetch result from pinot server. If we are seeing high amount of traffic to pinot broker with long request process latency, the whole pinot broker instances could be blocked due to limited amount of processing threads available to grizzly java web container. This change help to address the issue related to blocking call and ensure pinot broker have capacity to handle request even when some of the broker requests are taking too long

### CODE REVIEW

File: pinot-broker/src/main/java/org/apache/pinot/broker/api/resources/PinotClientRequest.java [hunk 1]
```diff
@@ -32,18 +32,25 @@
...
 import org.apache.pinot.common.metrics.BrokerMetrics;
 import org.apache.pinot.common.response.BrokerResponse;
+import org.apache.pinot.common.utils.CommonConstants;
 import org.apache.pinot.common.utils.CommonConstants.Broker.Request;
 import org.apache.pinot.spi.utils.JsonUtils;
+import org.glassfish.jersey.server.ManagedAsync;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import java.util.concurrent.TimeUnit;
```
Comment (Jackie-Jiang) [inline pinot-broker/src/main/java/org/apache/pinot/broker/api/resources/PinotClientRequest.java:52]: (format) re-order the imports

File: pinot-broker/src/main/java/org/apache/pinot/broker/api/resources/PinotClientRequest.java [hunk 2]
```diff
@@ -57,15 +64,17 @@
...
   @ApiOperation(value = "Querying pinot")
   @ApiResponses(value = {@ApiResponse(code = 200, message = "Query response"), @ApiResponse(code = 500, message = "Internal Server Error")})
-  public String processQueryGet(
+  public void processQueryGet(
       // Query param "bql" is for backward compatibility
       @ApiParam(value = "Query", required = true) @QueryParam("bql") String query,
       @ApiParam(value = "Trace enabled") @QueryParam(Request.TRACE) String traceEnabled,
-      @ApiParam(value = "Debug options") @QueryParam(Request.DEBUG_OPTIONS) String debugOptions) {
+      @ApiParam(value = "Debug options") @QueryParam(Request.DEBUG_OPTIONS) String debugOptions,
+      @Suspended final AsyncResponse asyncResponse) {
```
Comment (Jackie-Jiang) [inline pinot-broker/src/main/java/org/apache/pinot/broker/api/resources/PinotClientRequest.java:77]: (convention) remove the `final` key as we usually don't mark argument as final. Same for other methods.

File: pinot-broker/src/main/java/org/apache/pinot/broker/api/resources/PinotClientRequest.java [hunk 3]
```diff
@@ -76,24 +85,28 @@ public String processQueryGet(
...
   @Produces(MediaType.APPLICATION_JSON)
   @Path("query")
   @ApiOperation(value = "Querying pinot")
   @ApiResponses(value = {@ApiResponse(code = 200, message = "Query response"), @ApiResponse(code = 500, message = "Internal Server Error")})
-  public String processQueryPost(String query) {
+  public void processQueryPost(String query, @Suspended final AsyncResponse asyncResponse) {
     try {
       JsonNode requestJson = JsonUtils.stringToJsonNode(query);
+      if (requestJson.has("trace")) {
+        TimeUnit.SECONDS.sleep(15);
```
Comment (Jackie-Jiang) [inline pinot-broker/src/main/java/org/apache/pinot/broker/api/resources/PinotClientRequest.java:106]: What is this sleep for?

Comment (jamesyfshao) [reply pinot-broker/src/main/java/org/apache/pinot/broker/api/resources/PinotClientRequest.java]: ops, left over from my previous testing and it is supposed to be removed

**Pull request merged**